tests/js/server/utils-tests.js

```diff
@@ -0,0 +1,126 @@
+    });
+    return prom;
+};
+
+gpii.tests.express.user.utils.incorrectUsernamePassword = function (utils) {
+    var prom = utils.unlockUser("not-existing", "password");
+    prom.then(function () {
+        jqUnit.fail("This user shouldn't have unlocked with the supplied credentials.");
+    }, function () {
+        jqUnit.assert("Succeeded in not unlocking with incorrect credentials");
+    });
+    return prom;
+};
+
+// Each test has a request instance of `kettle.test.request.http` or `kettle.test.request.httpCookie`,
+// and a test module that wires the request to the listener that handles its results.
+fluid.defaults("gpii.tests.express.user.utils.caseHolder", {
+    gradeNames: ["gpii.test.webdriver.caseHolder"],
+    rawModules: [
+        {
+            name: "Testing login functions...",
+            tests: [
+                {
+                    name: "Create a new user and verify their data.",
+                    type: "test",
+                    sequence: [
+                        {
+                            task: "gpii.tests.express.user.utils.createUser",
+                            args: ["{gpii.express.user.utils}"],
+                            resolve: "jqUnit.assert",
+                            resolveArgs: ["Successfully created User"]
+                        }
+                    ]
+                },
+                {
+                    name: "Testing unlocking a user with correct credentials.",
+                    type: "test",
+                    sequence: [
+                        {
+                            task: "gpii.tests.express.user.utils.unlockUsernamePassword",
+                            args: ["{gpii.express.user.utils}"],
+                            resolve: "jqUnit.assert",
+                            resolveArgs: ["Successfully unlocked User"]
+                        }
+                    ]
+                },
+                {
+                    name: "Testing not unlocking a user with incorrect credentials.",
+                    type: "test",
+                    sequence: [
+                        {
+                            task: "gpii.tests.express.user.utils.incorrectUsernamePassword",
+                            args: ["{gpii.express.user.utils}"],
+                            reject: "jqUnit.assert",
+                            rejectArgs: ["Failed to open with bad credentials"]
+                        }
+                    ]
+                }
+
+            ]
+        }
+    ]
+});
+
+fluid.defaults("gpii.tests.express.user.utils.environment", {
+    gradeNames: ["gpii.test.express.user.environment"],
+    port:       8778,
+    pouchPort:  8764,
+    mailPort:   8725,
+    components: {
+        caseHolder: {
+            type: "gpii.tests.express.user.utils.caseHolder"
+        },
+        utils: {
+            type: "gpii.express.user.utils",
+            options: {
+                couch:  {
+                    userDbUrl: "http://127.0.0.1:8764/users"
```

Comment:
It would be better to populate the `userDbUrl` variable using an expander, so that you don't have to enter the port twice.  I typically use `fluid.stringTemplate` for this.

Reply:
this makes sense and is done